### PR TITLE
[New] Added test cases for record list add record menu

### PR DESCRIPTION
**Pull request by Srinivas9933**
Added test cases for record list add record menu component

- add-record-menu.spec.js

### CODE REVIEW

File: app/javascript/components/record-list/add-record-menu.spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  it("renders a single <Menu /> because user has access more than one module", () => {
+  
+    mountedComponent(<AddRecordMenu {...props} /> , state.merge(fromJS({ user: { modules: ["primerotest-1", "primerotest-3"] } })));
+
+    expect(screen.getByTestId('menu')).toBeInTheDocument()
+  
+  });
+
+  it("opens a <CreateRecordDialog /> if module allow_searchable_ids", () => {
```
Comment (dhernandez-quoin) [inline app/javascript/components/record-list/add-record-menu.spec.js:54]: This doesn't seem to test if the `CreateRecordDialog` is open.

File: app/javascript/components/record-list/container.spec.js [hunk 2]
```diff
@@ -309,83 +302,42 @@ describe("<RecordList />", () => {
...
-    expect(component.find(IndexTable)).to.have.length(1);
+    mountedComponent(<RecordList />,initialState, ["/cases"])
+    expect(screen.getByRole('toolbar')).toBeInTheDocument()
     done();
   });
 
   it("renders record view modal", done => {
-    expect(component.find(ViewModal)).to.have.lengthOf(1);
+    mountedComponent(<RecordList />,initialState, ["/cases"])
+    expect(screen.getAllByText('messages.record_list.rows_per_page')).toHaveLength(2)
```
Comment (dhernandez-quoin) [inline app/javascript/components/record-list/container.spec.js:313]: This doesn't seem to test if the `ViewModal` was rendered.

File: app/javascript/components/record-list/container.spec.js [hunk 3]
```diff
@@ -309,83 +302,42 @@ describe("<RecordList />", () => {
...
+   
     done();
   });
 
   it("opens the view modal when a record is clicked", () => {
-    component.find(TableCell).at(4).simulate("click");
-
-    expect(component.find(ViewModal).props().openViewModal).to.be.true;
+    mountedComponent(<RecordList />,initialState, ["/cases"])
+    expect(screen.getAllByText('messages.record_list.rows_per_page')).toHaveLength(2)
```
Comment (dhernandez-quoin) [inline app/javascript/components/record-list/container.spec.js:320]: This no longer performs a click and doesn't seem to test if the ViewModal was opened.

File: app/javascript/components/record-list/container.spec.js [hunk 4]
```diff
@@ -309,83 +302,42 @@ describe("<RecordList />", () => {
...
+        mountedComponent(<RecordList />,initialState.setIn(["application", "online"], false), ["/cases"])
+        expect(screen.getByTestId('filters')).toBeInTheDocument()
     });
   });
 
   describe("when age is 0", () => {
     it("renders a 0 in the cell ", () => {
-      expect(component.find(TableRow).at(1).find(TableCell).at(2).find(ConditionalWrapper).text()).to.equal("0");
+        mountedComponent(<RecordList />,initialState.setIn(["application", "online"], false), ["/cases"])
+        expect(screen.getAllByText('0-0 messages.record_list.of 0')).toHaveLength(2)
```
Comment (dhernandez-quoin) [inline app/javascript/components/record-list/container.spec.js:340]: Not sure why this is checking the pagination.

**Review by dhernandez-quoin — CHANGES_REQUESTED** [review]

File: app/javascript/components/record-list/container.spec.js (hunk 4 above)
Comment (Srinivas9933) [reply app/javascript/components/record-list/container.spec.js:340]: To check if the rows are empty

File: app/javascript/components/record-list/add-record-menu.spec.js (hunk 1 above)
Comment (Srinivas9933) [reply app/javascript/components/record-list/add-record-menu.spec.js]: Done

File: app/javascript/components/record-list/container.spec.js (hunk 2 above)
Comment (Srinivas9933) [reply app/javascript/components/record-list/container.spec.js]: Done

File: app/javascript/components/record-list/container.spec.js [hunk 5]
```diff
@@ -309,83 +302,42 @@ describe("<RecordList />", () => {
...
-      initialState.setIn(["application", "online"], false),
-      ["/cases"]
-    );
 
     it("when a record is clicked it does not open the view modal", () => {
-      offlineComponent.find(TableCell).at(3).simulate("click");
-
-      expect(component.find(ViewModal).props().openViewModal).to.be.false;
+        mountedComponent(<RecordList />,initialState.setIn(["application", "online"], false), ["/cases"])
+        expect(screen.getByTestId('filters')).toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (Srinivas9933) [reply app/javascript/components/record-list/container.spec.js]: Done

**Pull request merged**